backend/backend.js

```diff
@@ -1,142 +1,174 @@
     }
-    else if (job != undefined){
-        let result = findUserByJob(job);
-        result = {users_list: result};
-        res.send(result);
+});
+
+//ADD A USER
+app.post('/users', async (req, res) => {
+    try{
+        const user = req.body;
+        const result = await userServices.addUser(user);
+
+        if (result === undefined){
+            res.status(442).send('Unprocessable Entity');
+        }
+        else{
+            res.status(201).send(result);
+        }
+    } catch (error) {
+        console.log(error);
+        res.status(500).send('An error occured in the server');
     }
-    else{
-        res.send(users);
+});
+
+//SAVE AN EVENT
+app.post('/users/:id', async (req, res) => {
+    try{
+        eventId = req.body['eventId'];
+        hostId = req.params['id'];
+
+        const result = await userServices.saveEvent(hostId, eventId);
+        if (result === undefined || result === false){
+            res.status(442).send('Unprocessable Entity');
+        }
+        else{
+            res.status(201).send(result);
+        }
+    } catch (error) {
+        console.log(error);
+        res.status(500).send('An error occured in the server');
     }
 });
 
-const findUserByName = (name) =>{
-    return users['users_list'].filter( (user) => user['name'] === name);
-}
+//LOGIN AUTHENTICATION
+app.post('/login', async (req, res) => {
+    try{
+        const result = await userServices.validateUser(req.body);
+        if (result===true)
+            res.status(200).send('Successful login');
+        else    
+            res.status(401).end();
+    } catch (error) {
+        console.log(error);
+        res.status(500).send('An error occured in the server');
+    }
+});
 
-const findUserByJob = (job) =>{
-    return users['users_list'].filter( (user) => user['job'] === job);
-}
+//DELETE A USER
+app.delete('/users/', async (req, res) => {
+    const userToDel = await userServices.delUser(req.body);
+    res.send(userToDel);
+    // if (userToDel)
+    //     res.status(202).send(userToDel)
+    // else
+    //     res.status(500).send(userToDel)
+});
 
-const findUserbyJobName = (name, job) => {
-    return users['users_list'].filter( (user) => user['name'] === name && user['job'] == job);
-}
 
-app.get('/users/:id', (req, res) => {
-    const id = req.params['id'];
-    let result = findUserById(id);
-    if (result === undefined || result.length ==  0)
-        res.status(404).send('Resource not found');
-    else{
-        result = {users_list: result};
-        res.send(result);
+//ADD A FRIEND
+app.patch('/users/:id', async (req, res) => {
+    try{
+        const userId = req.params['id'];
+        const friendId = req.body.friendId;
+
+        const result = await userServices.addFriend(userId, friendId);
+        if (result === true){
+            res.status(200).send('Friend Successfully Added');
+        }
+        else if (result === false){
+            res.status(300).send('Users are already friends');
```

Comment:
if the user is already friends, the response might be better described by status 409 (conflict)

Reply:
Looks like it has already been changed